File: examples/pascal/constructs/comments.pas
// Ported from examples/lumen/constructs/comments.lm by scripts/port_examples.py; edit the Lumen original, not this file.
var x: integer;
var result: integer;
var value: integer;
var counter: integer;

function add_numbers(a: integer; b: integer): integer;
begin
    add_numbers := a + b;
end;

begin
    writeln('Test: Comments Support');
    x := 42;
    writeln(x);
    result := x * 2;
    writeln(result);
    value := add_numbers(10, 20);
    writeln(value);
    if value > 20 then begin
        writeln('Value is greater than 20');
    end else begin
        writeln('Value is 20 or less');
    end;
    counter := 0;
    while counter < 3 do begin
        writeln(counter);
        counter := counter + 1;
    end;
    writeln('Done testing comments');
end.

File: examples/swift/constructs/comments.swift
// Ported from examples/lumen/constructs/comments.lm by scripts/port_examples.py; edit the Lumen original, not this file.
print("Test: Comments Support")
let x = 42
print(x)
let result = x * 2
print(result)
func add_numbers(a: Int, b: Int) -> Int {
    return a + b
}

let value = add_numbers(a: 10, b: 20)
print(value)
if value > 20 {
    print("Value is greater than 20")
} else {
    print("Value is 20 or less")
}
var counter = 0
while counter < 3 {
    print(counter)
    counter = counter + 1
}
print("Done testing comments")
